ncm-network/src/main/perl/nmstate.pm

```diff
@@ -377,13 +393,13 @@ sub generate_nmstate_config
     #     next-hop-interface:
     #  and so on.
     my $routes = [];
+    push @$routes, @{$self->make_nm_route_absent($name)};
+    push @$routes, \%default_rt if scalar %default_rt;
```

Comment:
Bit weird that at line 396 you are pushing on a single entry array containing a hashref but here you are appending a hashref instead. What does @$routes actually expect in it?

Reply:
@$routes, it contains array of hashref.
so will contain values such as 
```
- next-hop-interface: bond0
  state: absent
- destination: 0.0.0.0/0
  next-hop-address: x.x.x.x
  next-hop-interface: bond0
```

Reply:
Makes sense, so I think that means the previous line should be:
push @$routes, $self->make_nm_route_absent($name);

otherwise you're trying to append an array of hashrefs not a hashref. I _suspect_ Perl flattens the array or otherwise does something weird so I t still works but I think it's better to not rely on that.

Reply:
I read `perldoc -f push` and it says that it appends each element in the second list to the first.  So line 390 is technically correct: the second arg must be a list. But I always recall using push like: `push @a, $b;` so I guess it works to have a scalar as the second argument. Perl is weird.